modules/dnn/src/layers/elementwise_layers.cpp

```diff
@@ -813,20 +813,173 @@ struct BaseDefaultFunctor : public BaseFunctor
     static const char* const ocl_kernel_name;
 };
 
-struct GeluFunctor : public BaseDefaultFunctor<GeluFunctor>
-{
-    typedef GeluLayer Layer;
+namespace pytorch_math {
+    /* Reference from PyTorch
+        https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/cpu/vec/vec512/vec512_float.h#L187-L218
+    */
+    constexpr float c_erf_coef0 = 0.3275911f;
+    constexpr float c_erf_coef1 = 1.061405429f;
+    constexpr float c_erf_coef2 = -1.453152027f;
+    constexpr float c_erf_coef3 = 1.421413741f;
+    constexpr float c_erf_coef4 = -0.284496736f;
+    constexpr float c_erf_coef5 = 0.254829592f;
+
+    inline float erf(float v) {
+        float t = 1.f / fma(fabs(v), c_erf_coef0, 1.f);
```

Comment:
please, use fabsf, fmaf, expf, otherwise compiler will use double precision for intermediate values

Reply:
Although cppreference shows that they are available since c++11,

- fabsf: https://en.cppreference.com/w/cpp/numeric/math/fabs
- expf: https://en.cppreference.com/w/cpp/numeric/math/exp

but turns out it is not. See https://stackoverflow.com/a/54028049/6769366.